includes/global-functions.php

```diff
@@ -203,3 +240,348 @@ function distributor_get_registered_data() {
 
 	return $distributor_registered_data;
 }
+
+/**
+ * Pre-distribute callback for media data.
+ * This is the default pre-distribute callback for media data, used when the type is set to 'media' in distributor_register_data().
+ *
+ * @param mixed $media_id The media ID to be processed before distribution.
+ * @return array The extra data of the media to be distributed to the target site.
+ */
+function distributor_media_pre_distribute_callback( $media_id ) {
+	if ( ! $media_id ) {
+		return array();
+	}
+
+	// Get the media data.
+	$media = get_post( $media_id );
+	if ( ! $media ) {
+		return array();
+	}
+
+	$media_url = wp_get_attachment_url( $media->ID );
+
+	return array(
+		'title'       => $media->post_title,
+		'caption'     => $media->post_excerpt,
+		'description' => $media->post_content,
+		'alt'         => get_post_meta( $media->ID, '_wp_attachment_image_alt', true ),
+		'url'         => $media_url,
+		'guid'        => $media->guid ?? $media_url,
+	);
```

Comment:
This could probably do with a mime type too, to allow for audio, video etc.

Reply:
Didn't got this. Do you mean to provide an option to restrict the media to only audio, video, images, or something similar?

Reply:
Sorry for the lack of clarity.

I mean the opposite: providing a mime type will expand the functionality to allow for audio, video and images. They're all uploaded as attachments and use the same data for the attachment URL. Providing the mime type (stored as `post_mime_type` in the posts table) will allow for developers to know the type of media they are dealing with.

Reply:
added in https://github.com/10up/distributor/pull/1309/commits/e26c01868d988e787cbb6675bccf5ddd276f3740